packages/react-components/react-dialog/src/components/Dialog/Dialog.types.ts

```diff
@@ -1,17 +1,80 @@
+import type * as React from 'react';
 import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
+import type { DialogContextValue } from '../../contexts/dialogContext';
 
 export type DialogSlots = {
-  root: Slot<'div'>;
+  /**
+   * Dimmed background of dialog.
+   * The default overlay is rendered as a `<div>` with styling.
+   * This slot expects a `<div>` element which will replace the default overlay.
+   * The overlay should have `aria-hidden="true"`.
+   */
+  overlay?: Slot<'div'>;
 };
 
-/**
- * Dialog Props
- */
-export type DialogProps = ComponentProps<DialogSlots>;
+export type DialogOpenChangeEvent = React.KeyboardEvent | React.MouseEvent | KeyboardEvent;
 
-/**
- * State used in rendering Dialog
- */
-// TODO: Add union of props to pick from DialogProps once they're implemented.
-// i.e. Required<Pick<DialogProps, 'property1' | 'property2'>>;
-export type DialogState = ComponentState<DialogSlots>;
+export type DialogOpenChangeData =
+  | { type: 'escapeKeyDown'; open: boolean; event: React.KeyboardEvent }
+  /**
+   * document escape keydown defers from internal escape keydown events because of the synthetic event API
+   */
+  | { type: 'documentEscapeKeyDown'; open: boolean; event: KeyboardEvent }
+  | { type: 'overlayClick'; open: boolean; event: React.MouseEvent }
+  | { type: 'triggerClick'; open: boolean; event: React.MouseEvent };
```

Comment:
Can you please clarify why `event` is passed in `data`?

```tsx
<Dialog onOpenChange={(event, data) => {
  // What `event` should I use?
  console.log(event, data.event)
}} />
```

IMO this looks redundant and confusing...